src/include/common/table_type.h

```diff
@@ -0,0 +1,15 @@
+#pragma once
+
+#include <cstdint>
+
+namespace kuzu {
+namespace common {
+
+enum class TableType : uint8_t {
```

Comment:
Why do we need to move TableType class to a separate file? Can we still keep this in the tableSchema.h?

Reply:
This enum is used in quite a lot places, distributed in binder, processor, storage, catalog, etc. So I think it's better to move it outside TableSchema to be more standalone.